services/system/AuthDelegate/plugin/src/lib.rs

```diff
@@ -1,25 +1,59 @@
 #[allow(warnings)]
 mod bindings;
+use std::str::FromStr;
+
+use bindings::*;
 mod errors;
 use errors::ErrorType::*;
+use exports::auth_delegate::plugin::api::{Error, Guest as Api};
+use exports::transact_hook_user_auth::{Claim, Guest as HookUserAuth, Proof};
+use transact::plugin::intf::add_action_to_transaction;
 
-// Other plugins
-use bindings::exports::transact_hook_user_auth::{Guest as HookUserAuth, *};
-use bindings::host::common::types as CommonTypes;
+use psibase::fracpack::Pack;
+use psibase::*;
 
 struct AuthDelegate;
 
+fn get_account_number(name: &str) -> Result<AccountNumber, Error> {
+    AccountNumber::from_str(name).map_err(|_| Error::from(InvalidAccountName))
+}
+
 impl HookUserAuth for AuthDelegate {
-    fn on_user_auth_claim(_account_name: String) -> Result<Option<Claim>, CommonTypes::Error> {
+    fn on_user_auth_claim(_account_name: String) -> Result<Option<Claim>, Error> {
         Err(NotYetImplemented("get_claims"))?
     }
 
     fn on_user_auth_proof(
         _account_name: String,
         _transaction_hash: Vec<u8>,
-    ) -> Result<Option<Proof>, CommonTypes::Error> {
+    ) -> Result<Option<Proof>, Error> {
         Err(NotYetImplemented("get_proofs"))?
     }
 }
 
+impl Api for AuthDelegate {
+    fn set_owner(owner: String) -> Result<(), Error> {
```

Comment:
This is unsafe without some kind of auth check.

Reply:
Of course, thanks - fixed